mimic-iii/buildmimic/duckdb/import_duckdb.py

```diff
@@ -0,0 +1,378 @@
+    make_concepts = args.make_concepts
+    mimic_code_root = args.mimic_code_root
+    skip_tables = args.skip_tables
+    skip_indexes = args.skip_indexes
+    #TODO: validate schema_name is valid identifier
+    schema_name = args.schema_name
+
+    if not skip_tables:
+
+        connection = duckdb.connect(output_db)
+        print("Connected to duckdb...")
+
+        try:
+            schema_prequel = ""
+            if schema_name is not None:
+                connection.execute(f"CREATE SCHEMA IF NOT EXISTS {schema_name};")
+                connection.execute(f"USE {schema_name};")
+                schema_prequel = f"USE {schema_name};"
+
+            print("Creating tables...")
+    
+            with open(os.path.join(mimic_code_root, 'mimic-iii','buildmimic','duckdb','duckdb_add_tables.sql'), 'r') as fp:
+                sql = fp.read()
+                connection.execute(sql)
+
+            print("Loading data...")
+            
+            for f in os.listdir(mimic_data_dir):
+                m = re.match(r'^(.*)\.csv(\.gz)*', f)
+                if m is not None:
+                    tablename = m.group(1).lower()
+                    tables_in_schema.add(tablename)
+                    tablename = tablename if schema_name is None else schema_name+'.'+tablename
+                    print(f"  {tablename}")
+                    connection.execute(f"COPY {tablename} from '{os.path.join(mimic_data_dir,m.group(0))}' (FORMAT CSV, DELIMITER ',', HEADER);")
+            
+            if not skip_indexes:
+                print("Adding indexes...")
+        
+                with open(os.path.join(mimic_code_root, 'mimic-iii','buildmimic','duckdb','duckdb_add_indexes.sql'), 'r') as fp:
+                    sql = fp.read()
+                    connection.execute(sql)
+
+            print("Running checks...")
+    
+            with open(os.path.join(mimic_code_root, 'mimic-iii','buildmimic','duckdb','duckdb_checks.sql'), 'r') as fp:
+                sql = fp.read()
+                result = connection.execute(sql).fetchall()
+                for row in result:
+                    print(f"{row[0]}: {row[2]} records ({row[1]} expected) - {row[3]}")
+
+        except Exception as error:
+            print("Failed setting up database: ", error)
+            raise error
+        finally:
+            if connection:
+                connection.close()
+                print("duckdb connection is closed")
+
+    #TODO: If both --schema-name and --skip-tables are specified, we won't have
+    # populated tables_in_schema with the data table names... so the views won't
+    # work... So, here, read the tables already in the destination schema from
+    # the DB and add those tablenames to tables_in_schema?
+
+    if make_concepts:
+        connection = duckdb.connect(output_db)
+        print("Connected to duckdb...")
+
+        print("Creating tables...")
+
+        # ccs_dx is an outlier...this is adapted from the BigQuery version...
```

Comment:
why is it an outlier?

Reply:
Yes, comment should probably be more specific. For the record, 1. because it's the only non-query-based table (that gets loaded with data) in the concepts build process, 2. the Postgres DB puts it in the public schema instead of `mimiciii`... which I am guessing is because it's also applicable to the MIMIC-IV concepts?